### PR TITLE
Merge the Jakarta NoSQL blogposts into draft

### PR DESCRIPTION
**Pull request by dearrudam**
Ref: https://github.com/OpenLiberty/blogs/issues/3536

It is following the recommendation from @GraceJansen - https://github.com/OpenLiberty/blogs/pull/3650#issuecomment-2022589975 - and @lauracowen - https://github.com/OpenLiberty/blogs/pull/3650#issuecomment-2039529276 - ! (Thanks for the support!)

## Changes:

- Split the original blog post into 3 parts:
  -  "Simplifying NoSQL Database Integration with Jakarta NoSQL"
  -  "Jakarta NoSQL in Action: JNopo Game"
  -  "Jakarta NoSQL in Action: Switching NoSQL Databases with Ease"

### CODE REVIEW

**Comment by dearrudam** [comment]
@fabio-franco , could you review it for me ?!

**Comment by fabio-franco** [comment]
Hi @dearrudam, sure! Sorry for the delay to reply. I have seen that for the draft it is not recommended to have the `blog_tags.json`, just for the next stage, the staging (https://github.com/OpenLiberty/blogs/tree/draft?tab=readme-ov-file). Is that ok for you to remove the `blog_tags.json` of this pr?

File: posts/2024-04-17-jakarta-nosql-in-action-meet-jnopo-game.adoc [hunk 1]
```diff
@@ -1046,143 +742,9 @@ cd sample-jakartaNoSQL-game
...
-----
-git clone https://github.com/OpenLiberty/sample-jakartaNoSQL-game.git
-cd sample-jakartaNoSQL-game
-cd database-switching-challenge
-----
-
 == Key Takeaways
 
-These challenges were an amazing opportunities to learn not just about Jakarta NoSQL and Jakarta Data to simplify the NoSQL integration with Java applications but be exploring new development approaches using interesting Java features.
+This challenge was an amazing opportunities to learn not just about Jakarta NoSQL and Jakarta Data to simplify the NoSQL integration with Java applications but be exploring new development approaches using interesting Java features.
```
Comment (fabio-franco) [inline posts/2024-04-17-jakarta-nosql-in-action-meet-jnopo-game.adoc]: Suggestion, change `opportunities` by `opportunity`

File: posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc [hunk 2]
```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: "Jakarta NoSQL in Action: Switching NoSQL Databases with Ease"
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/dearrudam
+author_github: https://github.com/dearrudam
+seo-title: TITLE - OpenLiberty.io
+seo-description: |
+  Nowadays, in the cloud era, switching between NoSQL solutions could be interesting to save resources and costs, but it could be a nightmare, specially if we're highly coupled to a particular vendor. It will cause some changes that will make this approach very expensive, like: time spent on the change, the learning curve of a new API to use with this database, the code that will be lost, the persistence layer that needs to be replaced, etc. This is where the flexibility of Jakarta NoSQL with an implementation like Eclipse JNoSQL together shines.
```
Comment (fabio-franco) [inline posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc]: Suggestion: change `specially` by `especially`

File: posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+// image::img/blog/FILE_NAME[IMAGE CAPTION ,width=70%,align="center"]
+// -------------------------
+// "[.img_border_light]" = This adds a faint grey border around the image to make its edges sharper. Use it around screenshots but not
+// around diagrams. Then double check how it looks.
+// There is also a "[.img_border_dark]" class which tends to work best with screenshots that are taken on dark backgrounds.
+// Change "FILE_NAME" to the name of the image file. Also make sure to put the image into the right folder which is: img/blog
+// change the "IMAGE CAPTION" to a couple words of what the image is
+// // // // // // // //
+
+Nowadays, in the cloud era, switching between NoSQL solutions could be interesting to save resources and costs, but it could be a nightmare, specially if we're highly coupled to a particular vendor. It will cause some changes that will make this approach very expensive, like: time spent on the change, the learning curve of a new API to use with this database, the code that will be lost, the persistence layer that needs to be replaced, etc.
```
Comment (fabio-franco) [inline posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc]: Same here: `specially` by `especially`

File: posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+
+NOTE: As of March 2024, Eclipse JNoSQL supports about https://www.jnosql.org/docs/supported_dbs.html[30 NoSQL databases].
+
+== Jakarta NoSQL in Action: Switching NoSQL Databases with Ease
+
+To explore this capability, let's use the link:../../../2024/04/17/jakarta-nosql-in-action-meet-jnopo-game.html[*JNopo* game].
+
+Here's the challenge: let's suppose that it's needed to change the *JNopo*'s Document NoSQL database, *MongoDB*, to another Document NoSQL database for save money purposes, e.g. *Couchbase*.
+
+By using Jakarta NoSQL specification to connect to the NoSQL solutions, we got some benefits and one of them is that with just a few changes, we'll be able to archive this goal easily.
```
Comment (fabio-franco) [inline posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc]: I guess `achive` would fit better than `archive` here

File: posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+Contributing to these projects is not just with codes, you could help a lot by promoting and speaking about them wherever you go!
+
+Contributing to open-source is a great way to boost your career, and improve your skills to become an effective developer and relevant in the market! Think about that!
+
+== Special Thanks
+
+I'm bursting with gratitude and would love to give a big shout-out to my incredible Java community friends for their unwavering support throughout my journey. A special round of applause for:
+
+- Otavio Santana, you're not just a mentor but a guiding star in my open-source journey. Your mentorship have opened doors for me to become an active open-source contributor and a proud Eclipse Foundation committer. Thank you for being such a monumental part of my journey. Also, thanks for your insightful reviews of the codes featured in this blog post.
```
Comment (fabio-franco) [inline posts/2024-04-20-jakarta-nosql-challenge-switching-nosql-easily.adoc]: Suggestion: maybe change `Your mentorship have...` to `Your mentorship has...`

File: posts/2024-04-15-simplifying-nosql-database-integration-with-jakarta-nosql.adoc [hunk 6]
```diff
@@ -0,0 +1,418 @@
...
+// // // // // // // //
+
+In the data-driven age of AI and cloud computing, choosing the right data storage solution directly impacts the success of a solution. We currently have two major data storage approaches to rely on: relational and NoSQL databases. Within the Java space, when it comes to working with relational databases, the Java Persistence API (JPA) specification has undoubtedly established itself as a robust and widely adopted API. In contrast, the crowded landscape of NoSQL database options opens a large set of integration options for developers, who instead of relying on standards, often find themselves learning and depending on vendor's proposed solutions.
+
+> "The way you define data storage and code design will either facilitate or hinder the project's maintainability, performance, scalability, and ability to change databases later on."
+
+Fortunately, the Java community adopting NoSQL storage solutions no longer needs to worry about these concerns. The Jakarta NoSQL specification offers a well-defined, standard-based implementation with low learning curve and familiar coding APIs, thanks to its similarities with the widely-adopted JPA specification.
```
Comment (fabio-franco) [inline posts/2024-04-15-simplifying-nosql-database-integration-with-jakarta-nosql.adoc]: Maybe consider changing from "implementation with low learning curve" to "implementation with a low learning curve"

File: posts/2024-04-15-simplifying-nosql-database-integration-with-jakarta-nosql.adoc [hunk 7]
```diff
@@ -0,0 +1,418 @@
...
+==== Challenges of NoSQL Integration
+
+Modern solution's requirements frequently requires the capabilities and benefits of different types of NoSQL databases, therefore, we should be able to work with multiple NoSQL solutions, coming from different vendors. Having said that, we can expect to face challenges such as:
+
+* A high cognitive load necessary when choosing a NoSQL database for their solutions;
+* A significant learning curve derived from usage of different database APIs;
+* Extra time invested on changing/maintaining existing codebase;
+* Potentially increase in complexity when onboarding new developers to the team;
+
+Furthermore, in the cloud era and the consumption-based pricing models, we must consider ways to deliver solutions with efficient resource consumption for costs reduction. This is how potential discussions may rise, opening possibility for considering changes of the underlying database used by existing applications.
```
Comment (fabio-franco) [inline posts/2024-04-15-simplifying-nosql-database-integration-with-jakarta-nosql.adoc]: Consider changing from "This is how potential discussions may rise," to "This is how potential discussions may arise,"

**Comment by dearrudam** [comment]
@fabio-franco, thanks for the feedback! I made the changes that you mentioned, could you review it again? 

@GraceJansen and @lauracowen, could you review it again!?

**Comment by fabio-franco** [comment]
Here is the link to review it in the draft area: https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/

**Comment by dearrudam** [comment]
@fabio-franco, I need to update the API to the latest version... I'll open a new PR with this upgrade!

**Pull request merged**